src/symbol/collision_index.js

```diff
@@ -317,41 +278,41 @@ class CollisionIndex {
             // Since there's no actual collision taking place, the circle vs. square
             // distinction doesn't matter as much, and box geometry is easier
             // to work with.
-            const projectedPoint = this.projectAndGetPerspectiveRatio(posMatrix, blocking.anchorPointX, blocking.anchorPointY);
-            const tileToViewport = textPixelRatio * projectedPoint.perspectiveRatio;
-            const x1 = blocking.x1 * tileToViewport + projectedPoint.point.x;
-            const y1 = blocking.y1 * tileToViewport + projectedPoint.point.y;
-            const x2 = blocking.x2 * tileToViewport + projectedPoint.point.x;
-            const y2 = blocking.y2 * tileToViewport + projectedPoint.point.y;
             const bbox = [
-                new Point(x1, y1),
-                new Point(x2, y1),
-                new Point(x2, y2),
-                new Point(x1, y2)
+                new Point(feature.x1, feature.y1),
+                new Point(feature.x2, feature.y1),
+                new Point(feature.x2, feature.y2),
+                new Point(feature.x1, feature.y2)
```

Comment:
Does the bbox polygon need five points (with the last point being the same as the first) in order for the intersection test to work properly?

Reply:
I don't think so? I actually haven't looked closely at how `intersection_tests.js` implements this, but it's the same behavior as before, right?

Reply:
Yep, it's the same behaviour as before.

https://github.com/mapbox/mapbox-gl-js/blob/13ea16df5eabc4c8867d69e81da180cdcd6cc6b2/src/util/intersection_tests.js#L112-L124

`lineIntersectsLine`, which is used by `polygonIntersectsPolygon`, *seems* to expect a repeated vertex when used this way (the last segment is never tested). BUT I think it's still (accidentally?) correct, because it's impossible for this missing edge to be crossed without either crossing a different edge or having a point within the polygon. So, I'm not sure if this a bug or not

Reply:
🙈 Do you have any ideas for tests or assertions we could put in to make it clear what types of geometry are supported? I haven't wrapped my head around it and for now my assumption is it's OK.